src/create.rs

```diff
@@ -520,18 +584,65 @@ impl Execution {
 
         // Compile the installer
         info!("Compiling the installer");
-        let mut compiler = self.compiler()?;
+
+        let mut compiler = if !self.wix_toolset.is_modern() {
```

Comment:
Change to `if self.wix_toolset.is_legacy()` or swap the cases to avoid using the negate.